.github/workflows/check-public-api.yaml

```diff
@@ -0,0 +1,59 @@
+name: Check public API
+
+env:
+  RUST_BACKTRACE: 1
+on:
+  push:
+    tags:
+      # this is _not_ a regex, see: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet
+      - '[0-9]+.[0-9]+.[0-9]+*'
+  workflow_dispatch:
+    inputs:
+      diff-api:
+        required: false
+        description: 'Diff API'
+jobs:
+  setup:
+    name: Set up
+    runs-on: ubuntu-latest
+    outputs:
+      VERSION: ${{ steps.setup.outputs.VERSION }}
+      SHOULD_CHECK_API: ${{ steps.setup.outputs.SHOULD_CHECK_API }}
+    steps:
+
+      - name: Set up env vars
+        id: setup
+        shell: bash
+        run: |
+          VERSION=${GITHUB_REF/refs\/tags\//}
+          echo ::set-output name=VERSION::${VERSION}
+          SHOULD_CHECK_API=$(echo $VERSION | grep -c '^[0-9]\+\.[0-9]\+\.[0-9]\+\(-\([a-zA-Z]\+\)\?[0-9]*\)\?$' || true)
+          echo ::set-output name=SHOULD_CHECK_API::${SHOULD_CHECK_API}
+          echo $VERSION
+          echo $SHOULD_CHECK_API
+
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install latest nightly
+        uses: actions-rs/toolchain@v1
+        with:
+            toolchain: nightly
+            profile: minimal
+            override: true
+
+      - name: Generate API change report
+        run: |
+          cargo install cargo-public-api
+          cargo install ripgrep
+          LATEST_VERSION=$(cargo search wasmer | rg --only-matching --replace '$1' 'wasmer = "(.*)"')
+          cargo public-api --manifest-path=lib/api/Cargo.toml --diff-git-checkouts $LATEST_VERSION master > diff.txt
```

Comment:
Maybe make `LATEST_VERSION` a workflow input? And if it's not specified make the default the last tagged release, which you can get with `git tag --list | tail -n 1` which doesn't make `rg` necessary.